### PR TITLE
Diferentiate, integrate and evaluate with symbols

### LINKED ISSUE #148: Allow differentiation and integration with respect to symbolic variable names
```jl
t, x = set_variables("x y")
integrate(t, :t)
integrate(x, :t)
```

### LINKED ISSUE #151: Partial `evaluate`ion
Suppose I have a Taylor series `s` in 3 variables `x, y, z`.
I need to be able to evaluate `s` for some numerical value of `x`, and obtain as response a TaylorSeries in the two variables `y and z`. 

(It can still be a 3-variable series; it just won't have any monomials including any power of `x`.)

### PR DESCRIPTION
**Pull request by lbenet**
This fixes #148 and fixes #151

EDIT: Correct spelling

### CODE REVIEW

**Comment by lbenet** [comment]
I'm not sure if a rebase is needed here...

**Comment by dpsanders** [comment]
Yes I think it needs to be rebased. Should be trivial

**Comment by lbenet** [comment]
I also thought so, but github thinks it can be merged directly. In any case, it is ready for review and also to be merged.

**Comment by dpsanders** [comment]
Note that I just rebased against master.

File: src/calculus.jl [hunk 1]
```diff
@@ -131,6 +131,8 @@ function derivative(a::HomogeneousPolynomial, r::Int)
 
     return HomogeneousPolynomial{T}(coeffs, a.order-1)
 end
+derivative(a::HomogeneousPolynomial, s::Symbol) =
+    derivative(a, findfirst(get_variable_symbols(), s))
```
[the remark this answers is not in the document]
Comment (lbenet) [reply src/calculus.jl]: So the proposal is to have something like `lookupsymb(s)` which is equivalent to `findfirst(get_variable_symbols(), s)`. Is this what you mean?

[the remark this answers is not in the document]
Comment (dpsanders) [reply src/calculus.jl]: Yes exactly. Maybe lookupvar

[the remark this answers is not in the document]
Comment (lbenet) [reply src/calculus.jl]: Ok.

**Comment by lbenet** [comment]
Just pushed another commit following your suggestions

**Comment by lbenet** [comment]
Incidentally, I did not export `lookupvar`; do you think it is needed?

**Comment by dpsanders** [comment]
I think it's fine (or better) if it's not exported.

**Comment by dpsanders** [comment]
LGTM, thanks!

**Comment by dpsanders** [comment]
Thanks!

**Comment by lbenet** [comment]
Thanks a lot!

**Pull request merged**